client/app/services/group/currentGroup.service.js

```diff
@@ -14,7 +14,18 @@ export default class CurrentGroup {
     Object.assign(this, {
       value: {},
       User,
-      SessionUser
+      SessionUser,
+      map: {
```

Comment:
The interface of the map proxy methods is quite different from the `Current*` services, which essentially are just data caches.
Somehow this looks to me like an "observer" scheme. The "groupMap" component registers an observer by setting the functions. So there are typical corner cases, e.g. what happens when the `groupMap` component gets destroyed? Will the proxy methods get set to doing nothing again?

As far as I understood, developers in Angular1 should avoid custom observer patterns and prefer watchable data bindings. It should be easier to understand when everything follows a similar pattern.

In this case, we could try to add a `mapOptions` object (e.g. to the currentGroup service)
It can get watched in `groupMap` like this:

```
this.$scope.$watch(() => this.CurrentGroup.mapOptions, () => {
    // run update methods
});
```

From other places in the code, you could set `currentGroup.mapOptions.overview = true`.

Does this make sense?